Watchman.Discord/WatchmanBot.cs

```diff
@@ -105,6 +105,16 @@ public WorkflowBuilder GetWorkflowBuilder()
                         .AddHandler(this.PrintDebugExceptionInfo, onlyOnDebug: true)
                         .AddHandler(this.PrintExceptionOnConsole);
                 })
+                .AddOnUserAddedReaction(builder =>
+                {
+                    builder
+                        .AddFromIoC<ReactionsService>(service => reactionContext => service.OnUserAddedReaction(reactionContext));
```

Comment:
to cały czas wygląda jak krzywy hack

ogarnianie reakcji powinno być po stronie frameworka, skoro to on je ogarnia
Watchman.Discord powinien dostawać już gotowe informacje do kontrolerów, bez konieczności dodatkowego kombinowania z własnym serwisem

Reply:
Nie wiem o co ci chodzi. Przecież używa się tego tak samo jak innych handlerów. Dla mnie ten kod wygląda dobrze, wiadomo o co w nim chodzi, jest prosty i trzyma się standardu (handlerów). 

Chyba, że chodzi ci o to, że delegat serwisu jest wywoływany zamiast po prostu przekazany do frameworka. Niestety musi tak być bo inaczej nie będzie działało.

Reply:
Wyobraź sobie sytuacje gdzie chcemy postawić innego bota na tym frameworku 

Wtedy nie powinniśmy pisać funkcjonalności, które powinny być w frameworku

Rzeczy które aktualnie są poza frameworkiem to rzeczy, które nie powinny w nim być - bo są związane z tym jak ma działać akurat Watchman

Reply:
To jak chcesz przekazać informacje o serwisie i jego delegacie bez wysyłania odpowiednich danych do frameworka? Chyba po to jest AddFromIoC aby podać frameworkowi jakoś zależność i tą zależność wykorzystać do dalszej obsługi. Zresztą mateusz mi zaproponował taki zapis (w zasadzie go trochę zmieniłem, ale jest tak naprawdę tak jak mateusz podał). 

"Rzeczy które aktualnie są poza frameworkiem to rzeczy, które nie powinny w nim być - bo są związane z tym jak ma działać akurat Watchman" - to w takim razie czemu w naszym kodzie framework ma zależności takie jak: HelpDBGeneratorService, UnmutingExpiredMuteEventsService czy ResponsesInitService? Te zależności są poza frameworkiem.

"Wyobraź sobie sytuacje gdzie chcemy postawić innego bota na tym frameworku" - normalnie się podaje odpowiednią klasę i jej delegat tak jak w przypadku ReactionService.

Reply:
UnmutingExpiredMuteEventsService - bo to jest funkcjonalność tego konkretnego bota
ResponsesInitService, HelpDBGeneratorService - bo to są funkcjonalności bota które idą obok kontrolerów a nie przez kontrolery, a muszą być obsługiwane 

ReactionService mógłby iść przez WorkflowBuildera, gdyby miał akcje które mają sie wykonywać, a są związane typowo z botem - np muszą operować na domenie bota 

`normalnie się podaje odpowiednią klasę i jej delegat tak jak w przypadku ReactionService` - no tylko po co robić to samo za każdym razem? sens frameworka jest taki, że przy stawianiu na nim kolejnego bota, nie chcemy klepać rzeczy, które powinny nim być 

no chyba że są to rzeczy, które w każdym z botów mogą zachowywać się inaczej - tak jak końcowa obsługa wyjątków

Reply:
"UnmutingExpiredMuteEventsService - bo to jest funkcjonalność tego konkretnego bota" - a ReactionService nie jest?

"ReactionService mógłby iść przez WorkflowBuildera" - a wcześniej mówiłeś, że rzeczy nie związane z frameworkiem nie powinny być w nim. Mam naprawdę teraz mętlik w głowie...

"normalnie się podaje odpowiednią klasę i jej delegat tak jak w przypadku ReactionService - no tylko po co robić to samo za każdym razem? [...]" - mateusz mi zaproponował takie rozwiązanie aby przez AddFromIoC dać frameworkowi ReactionService i w argumencie metody dać delegat serwisu (choć ja musiałem jeszcze przekazywanie odwołania do delegatu zamienić na wywołanie, bo inaczej program w innym miejscu wywalałby wyjątki).

Reply:
wszystko zależy od tego co ReactionsService ma robić
tutaj nie chodzi o nazwe klasy tylko o jej zastosowanie 

zauważ że do obsługi komend nie podpinamy niczego, bo to mechanizm który jest typowo frameworkowy 
tak samo jak obsługa reakcji 

za to obsługa mutowania to funkcjonalność typowo watchmana